packages/readmodel/test/eserviceAggregator.test.ts

```diff
@@ -136,4 +140,134 @@ describe("E-service aggregator", () => {
       metadata: { version: 1 },
     });
   });
+
+  /**
+   * !! IMPORTANT !!
+   * This test ensures that the order of attribute groups are consistent with the original order
+   * in the descriptor, even after protobuf deserialization, splitting,
+   * aggregating a reserialization of the eservice object.
+   *
+   * This is important because some functionalities in APIV2 (e.g. in-add attributes endpoints)
+   * rely on the order of these groups.
+   */
+  it("should keep the descriptor attributes group order by groupIndex", () => {
+    const descriptorCertifiedAttributeGroups: EserviceAttributes["certified"] =
+      [
+        [getMockEServiceAttribute()],
+        [getMockEServiceAttribute(), getMockEServiceAttribute()],
+        [getMockEServiceAttribute(), getMockEServiceAttribute()],
+        [
+          getMockEServiceAttribute(),
+          getMockEServiceAttribute(),
+          getMockEServiceAttribute(),
+        ],
+      ];
+
+    const descriptorVerifiedAttributeGroups: EserviceAttributes["verified"] = [
+      [getMockEServiceAttribute()],
+      [getMockEServiceAttribute(), getMockEServiceAttribute()],
+      [getMockEServiceAttribute(), getMockEServiceAttribute()],
+      [
+        getMockEServiceAttribute(),
+        getMockEServiceAttribute(),
+        getMockEServiceAttribute(),
+      ],
+    ];
+
+    const descriptorDeclaredAttributeGroups: EserviceAttributes["declared"] = [
+      [getMockEServiceAttribute()],
+      [getMockEServiceAttribute(), getMockEServiceAttribute()],
+      [getMockEServiceAttribute(), getMockEServiceAttribute()],
+      [
+        getMockEServiceAttribute(),
+        getMockEServiceAttribute(),
+        getMockEServiceAttribute(),
+      ],
+    ];
+
+    const descriptor: Descriptor = {
+      ...getMockDescriptor(),
+      attributes: {
+        certified: descriptorCertifiedAttributeGroups,
+        verified: descriptorVerifiedAttributeGroups,
+        declared: descriptorDeclaredAttributeGroups,
+      },
+    };
+
+    const eservice = getMockEService(undefined, undefined, [descriptor]);
+    const serialized = EServiceAddedV2.toBinary({
+      eservice: toEServiceV2(eservice),
+    });
+
+    // eslint-disable-next-line functional/no-let
+    let deserialized = EServiceAddedV2.fromBinary(serialized).eservice;
```

Comment:
I would do this serialization and deserialization inside the for loop, so that it's also part of the repeated test.